src/inmanta/agent/handler.py

```diff
@@ -145,15 +153,47 @@ def wrapper(self: HandlerAPI[TResource], *args: object, **kwds: object) -> objec
             def bound(**kwds: object) -> object:
                 return f(self, **kwds)
 
+            _evict_after_last_access: float = evict_after_last_access
+            _evict_after_creation: float = evict_after_creation
+
+            # If legacy param `timeout` is set, it overrides `evict_after_creation`
+            if timeout and timeout > 0:
+                _evict_after_creation = timeout
+
+            # Legacy `for_version` parameter is used, compute
+            # evict_after_last_access and evict_after_creation
+            if for_version is not None:
+                if for_version:
+                    _evict_after_creation = 0
+
+                    if evict_after_last_access > 0:
+                        _evict_after_last_access = evict_after_last_access
+                    else:
+                        _evict_after_last_access = 60
+                else:
+                    _evict_after_last_access = 0
+
+                    # Successive overrides with a 5000s default
+                    _evict_after_creation = 5000
+                    if evict_after_creation > 0:
+                        _evict_after_creation = evict_after_creation
+                    if timeout and timeout > 0:
+                        _evict_after_creation = timeout
```

Comment:
I think it would be helpful to warn people that an override will happen if they use both versions.